src/jsMain/js/app/Menu/components/ColorPicker/index.jsx

```diff
@@ -1,52 +1,212 @@
+import chroma from 'chroma-js';
+import throttle from 'lodash/throttle';
+import Draggable from 'react-draggable';
+import { rgb2xy, xy2polar, xy2rgb } from '../../../../utils/colorUtils';
 
 const baaahs = sparklemotion.baaahs;
 
+const pickerRadius = 12;
+
 class ColorPicker extends Component {
   constructor(props) {
     super(props);
 
+    const { gadget } = this.props;
+    const { redI, greenI, blueI } = gadget.color;
+    const color = [redI, greenI, blueI];
+
     this.state = {
-      gadget: props.gadget,
+      gadget,
+      colors: [color],
+      radius: 200,
+      selectedIndex: -1,
+      grabbingIndex: -1,
+      clientWidth: 400,
     };
 
-    props.gadget.listen({onChanged: () => {this.forceUpdate()}});
+    // Send color updates once every 150ms while the picker is dragged
+    this._throttledHandleColorChange = throttle(this._handleColorChange, 150);
+
+    this._changeListener = {
+      onChanged: (gadget) => {
+        const { colors } = this.state;
+        const { redI, greenI, blueI } = gadget.color;
+        const color = [redI, greenI, blueI];
+
+        const updatedColors = colors.slice();
+        updatedColors[0] = color;
+
+        this.setState({ colors: updatedColors });
+      },
+    };
+    gadget.listen(this._changeListener);
+  }
+
+  componentWillUnmount() {
+    const { gadget } = this.state;
+    if (this._changeListener) {
+      gadget.unlisten(this._changeListener);
+    }
+  }
+
+  updateWheel() {
+    const { radius } = this.state;
+    if (!this._canvasEl || !radius) return;
+
+    const width = radius * 2;
+    const height = radius * 2;
+    const featheringPx = 1;
+
+    let ctx = this._canvasEl.getContext('2d');
+    let image = ctx.createImageData(width, height);
+    let data = image.data;
+
+    for (let x = -radius; x < radius; x++) {
+      for (let y = -radius; y < radius; y++) {
+        let [r] = xy2polar(x, y);
+        // Figure out the starting index of this pixel in the image data array.
+        let rowLength = 2 * radius;
+        let adjustedX = x + radius; // convert x from [-radius, radius] to [0, radius*2] (the coordinates of the image data array)
+        let adjustedY = y + radius; // convert y from [-radius, radius] to [0, radius*2] (the coordinates of the image data array)
+        let pixelWidth = 4; // each pixel requires 4 slots in the data array
+        let index = (adjustedX + adjustedY * rowLength) * pixelWidth;
+
+        let [red, green, blue] = xy2rgb(x, y, radius);
+        let alpha = 255;
+        if (r >= radius - featheringPx) {
+          alpha = Math.max(
+            255 - 255 * ((r - (radius - featheringPx)) / featheringPx),
+            0
+          );
+        }
+
+        data[index] = red;
+        data[index + 1] = green;
+        data[index + 2] = blue;
+        data[index + 3] = alpha;
+      }
+    }
+
+    ctx.putImageData(image, 0, 0);
   }
 
-  handleColorChange = ({ hex }) => {
+  _handleColorChange = () => {
```

Comment:
This is totally just an opinion and super minor and not something we actually need to change. 

I've noticed a convention that some JS devs preface methods that are meant to be "private" with an underscore.

I think I set babel up in this repo to actually allow this tc39 stage 3 proposal for private methods and fields: https://github.com/tc39/proposal-class-fields#private-fields

It's weird syntax because we're replacing `_` with `#`, so `_handleColorChange` becomes `#handleColorChange`, and to access this method we use `this.#handleColorChange`.

We don't need to use it if we think it's weird. It's only a stage 3 proposal. The only difference is it actually keeps these methods private/internal to this class. 

> To make fields private, just give them a name starting with #.

> By defining things which are not visible outside of the class, ESnext provides stronger encapsulation, ensuring that your classes' users don't accidentally trip themselves up by depending on internals, which may change version to version.

Reply:
Ahh, wow! I haven't heard of the whole `#` private proposal before. That's sorta crazy and cool and I'm down to try it. I've always used `_` as just a convention. I'll give `#` a shot 👍

Reply:
I tried this out and ran into a few issues: my IntelliJ gets a little confused by the #s and I ran into an error saying it's enabled even though it is (maybe this https://github.com/babel/babel/issues/8484 ?)

I switched it back to underscores for now, but I'd be down to try it again once the proposal is a bit more mature!